synapse/rest/media/v1/media_storage.py

```diff
@@ -145,15 +145,15 @@ async def finish() -> None:
                     f.flush()
                     f.close()
 
-                    spam = await self.spam_checker.check_media_file_for_spam(
+                    spam_check = await self.spam_checker.check_media_file_for_spam(
                         ReadableFileWrapper(self.clock, fname), file_info
                     )
-                    if spam:
+                    if isinstance(spam_check, Codes):
                         logger.info("Blocking media due to spam checker")
                         # Note that we'll delete the stored media, due to the
                         # try/except below. The media also won't be stored in
                         # the DB.
-                        raise SpamMediaException()
+                        raise SpamMediaException("Not found", spam_check)
```

Comment:
```suggestion
                        raise SpamMediaException(errcode=spam_check)
```

Since `msg` already defaults to `Not Found` on `SpamMediaException`.